test/fetch-mocked.test.js

```diff
@@ -136,20 +155,173 @@ describe('unit tests on fetch functions', function() {
   });
 
   describe('getAllowedDatabases', function() {
-    it('should return the correct results');
-    // ...
+    var results = {};
+    beforeEach(function() {
+      // No db commands are called here.
+      results.db = makeMockDB(null, null);
```

Comment:
if `db` is not needed you don't need to pass it in. I'd say lets remove the `beforeEach` function completely here so that we test under the most realistic circumstances.